python/triton/config.py

```diff
@@ -0,0 +1,310 @@
+    path: str
+    version: str
+
+
+def get_nvidia_tool(binary: str) -> NvidiaTool:
+    binary += sysconfig.get_config_var("EXE")
+    # nvidia backend root
+    paths = [
+        getenv(f"TRITON_{binary.upper()}_PATH"),
+        os.path.join(
+            os.path.dirname(__file__),
+            "backends",
+            "nvidia",
+            "bin",
+            binary,
+        )
+    ]
+    for path in paths:
+        if not path or not os.access(path, os.X_OK):
+            continue
+        try:
+            result = subprocess.check_output([path, "--version"], stderr=subprocess.STDOUT)
+            if result is None:
+                continue
+            version = re.search(r".*release (\d+\.\d+).*", result.decode("utf-8"), flags=re.MULTILINE)
+            if version is None:
+                continue
+            return NvidiaTool(path, version.group(1))
+        except subprocess.CalledProcessError:
+            pass
+
+    raise RuntimeError(f"Cannot find {binary}")
+
+
+class build:
+    """Configuration controlling how the native compiler is invoked"""
+    cc: env_str = env_str("CC")
+    backend_dirs: env_str_set = env_str_set("TRITON_CUDACRT_PATH", "TRITON_CUDART_PATH")
+
+
+class redis:
+    key_format: env_strd = env_strd("TRITON_REDIS_KEY_FORMAT", "triton:{key}:{filename}")
+    host: env_strd = env_strd("TRITON_REDIS_HOST", "localhost")
+    port: env_int = env_int("TRITON_REDIS_PORT", 6379)
+
+
+class cache:
+    dump_dir: env_strd = env_strd("TRITON_DUMP_DIR", lambda: get_triton_dir("dump"))
+    override_dir: env_strd = env_strd("TRITON_OVERRIDE_DIR", lambda: get_triton_dir("override"))
+    dir: env_strd = env_strd("TRITON_CACHE_DIR", lambda: get_triton_dir("cache"))
+
+    manager_class: env_class[CacheManager] = env_class("TRITON_CACHE_MANAGER", "CacheManager")
+    remote_manager_class: env_class[RemoteCacheBackend] = env_class("TRITON_REMOTE_CACHE_BACKEND", "RemoteCacheBackend")
+
+
+class compilation:
+    override: env_bool = env_bool("TRITON_KERNEL_OVERRIDE")
+    dump_ir: env_bool = env_bool("TRITON_KERNEL_DUMP")
+    store_binary_only: env_bool = env_bool("TRITON_STORE_BINARY_ONLY")
+    always_compile: env_bool = env_bool("TRITON_ALWAYS_COMPILE")
+    use_ir_loc: env_bool = env_bool("USE_IR_LOC")
+    enable_asan: env_bool = env_bool("TRITON_ENABLE_ASAN")
+    disable_line_info: env_bool = env_bool("TRITON_DISABLE_LINE_INFO")
+    frontend_debugging: env_bool = env_bool("TRITON_FRONT_END_DEBUGGING")
+    allow_non_constexpr_globals: env_bool = env_bool("TRITON_ALLOW_NON_CONSTEXPR_GLOBALS")
+
+
+class autotuning:
+    cache: env_bool = env_bool("TRITON_CACHE_AUTOTUNING")
+    print: env_bool = env_bool("TRITON_PRINT_AUTOTUNING")
+
+
+class LaunchHook(Protocol):
+
+    def __call__(self, metadata: LazyDict) -> None:
+        ...
+
+
+# This is of the form [attr_name, attr_val]
+# TODO: Use tuple instead of list for better typing.
+KernelAttr = list[str | int]
+
+
+class JitHookCompileInfo(TypedDict):
```

Comment:
`JitHookCompileInfo` -> `JITHookCompileInfo`

Reply:
We use `JITFunction` instead of `JitFunction`

Reply:
Oh right, good catch!